### ISSUE TITLE
Ability to discard changes while database is open

### ISSUE THREAD

**Issue by grenzor** [keepassxreboot/keepassxc#1312]
From what I can see changes can only be discarded by closing the database and choosing discard changes then. This can be a hassle to close and reopen just to discard changes and it would be better to also have an option+shortcut to discard changes while the database is open with something that would operate similar to CTRL+Z.

**Comment by sergeevabc** [comment]
Crucial feature indeed.

**Comment by droidmonkey** [comment]
Need to implement transactional database changes first. Then Undo/Redo. Then this will be a reality.

**Comment by boldandbusted** [comment]
Howdy. Just fatfingered the deletion of an entry as soon as I switched to the KeePassXC window. I have "Automatically save on Every Change" on. Don't even know what I deleted, just saw the rows move up... :( Would love to be able to undo oopses... Thanks.

**Comment by droidmonkey** [comment]
Likely in your recycle bin of the database, hope you had that enabled. You should also have backups of your database.

**Comment by boldandbusted** [comment]
> Likely in your recycle bin of the database, hope you had that enabled. You should also have backups of your database.

YES. It was in the "Recycle Bin" - I forgot that existsed. :) Thank you. And, I do have backups - I should have considered that as well. Cheers.

**Comment by ferenczy** [comment]
Are there any updates regarding this feature, please? I noticed that the milestone is set to version 2.8.0.

**Comment by phoerious** [comment]
No, but you can restore a previous state from the entry history.
